baybe/utils/sampling_algorithms.py

```diff
@@ -70,3 +76,58 @@ def farthest_point_sampling(
         remaining_point_indices.remove(selected_point_index)
 
     return selected_point_indices
+
+
+class DiscreteSamplingMethod(Enum):
+    """Available discrete sampling methods."""
+
+    Random = "Random"
+    """Random Sampling."""
+
+    FPS = "FPS"
+    """Farthest point sampling."""
+
+
+def sample_numerical_df(
+    df: pd.DataFrame,
+    n_points: int,
+    *,
+    method: DiscreteSamplingMethod = DiscreteSamplingMethod.Random,
+) -> pd.DataFrame:
+    """Sample data points from a numerical dataframe.
+
+    If the requested amount of points is larger than the number of available points,
+    the entire dataframe will be returned as many times at it fits into the requested
+    number and the specified sampling method will only return the remainder of points.
+
+    Args:
+        df: Dataframe with purely numerical entries.
+        n_points: Number of points to sample.
+        method: Sampling method.
+
+    Returns:
+        The sampled points.
+
+    Raises:
+        TypeError: If the provided dataframe has non-numerical content.
+        ValueError: When an invalid sampling method was provided.
+    """
+    if any(df[col].dtype.kind not in "iufb" for col in df.columns):
+        raise TypeError(
+            "'sample_numerical_df' only supports purely numerical data frames."
+        )
+
+    # Split points in trivial and sampled parts
+    n_trivial = n_points // len(df)
+    n_sampled = n_points - n_trivial * len(df)
+
+    ilocs = list(range(len(df))) * n_trivial
+    if n_sampled > 0:
+        if method is DiscreteSamplingMethod.FPS:
+            ilocs += farthest_point_sampling(df.values, n_sampled)
+        elif method is DiscreteSamplingMethod.Random:
+            ilocs += df.sample(n_sampled).reset_index(drop=True).index.tolist()
```

Comment:
I think this line is just wrong, could that be? It should add the positional indices of the sampled points, but you are resetting the index after sampling, so you are completely losing the positional information.

Reply:
good catch, resetting the index needs to go before the sampling

Reply:
fixed